spec/core/elastic_connector_actions_spec.rb

```diff
@@ -0,0 +1,558 @@
+require 'core/elastic_connector_actions'
+require 'connectors/connector_status'
+require 'connectors/sync_status'
+require 'utility/es_client'
+
+describe Core::ElasticConnectorActions do
+  let(:connector_id) { 'one-two-three' }
+  let(:es_client) { double }
```

Comment:
Not for this PR, but has there been any discussion about getting a local Elasticsearch available on the CI instances for connectors? That would make these more "integration" tests than unit tests, but also might significantly lower the mocking burden.

Reply:
We do have integration test that is ran nightly - you can run it with `make ftest` - it's a functional test though, not an integration one.

We did not have a discussion about that and it's something nice to talk about indeed.

IMO there's not much mocking burden as long as the class doesn't do too much - and mocking helps see class dependencies on 3rd-party components when looking at the test, which I think is quite valuable too. 

All in all, I think the strategy I personally like the most is - as many as possible unit tests + a small number of integration tests that test big chunks of the system + a small number of functional tests that test high-level scenarios.